hail/src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -170,31 +157,6 @@ final case class ApplyBinaryPrimOp(op: BinaryOp, l: IR, r: IR) extends IR
 final case class ApplyUnaryPrimOp(op: UnaryOp, x: IR) extends IR
 final case class ApplyComparisonOp(op: ComparisonOp[_], l: IR, r: IR) extends IR
 
-object MakeArray {
```

Comment:
MakeArray.unify started breaking as soon as I modified PruneDeadFields.upcast to not check type equality, which I did because prior to this change, PruneDeadFields.upcast had 2 functionalities, the first to loosen requiredeness, the second to change other aspects of the type (such as the fields of a TStruct, which was the behavior that had triggered the python test failures when I removed If.unify).

The simplest change here is just to remove MakeArray, because outside of tests, it was only used when the "typ" argument wasn't passed (Parser.scala), which means the former behavior (type requiredeness relaxation) was the only use.

Reply:
decided to save this for a 2nd pr